lib/puppet/parser/ast/pops_bridge.rb

```diff
@@ -73,6 +73,9 @@ class Program < Puppet::Parser::AST::TopLevelConstruct
     def initialize(program_model, context = {})
       @program_model = program_model
       @context = context
+      # @todo lutter 2015-03-10: we maintain a list of definitions we
+      # instantiated so that we can attach CapabilityMappings to them
+      @types = {}
```

Comment:
Yeah, I agree that the way it currently is, you won't be able to have cross-file capability annotations.

I am not sure what the right approach is - `PopsBridge::Program#instantiate` must return an array of `Puppet::Resource::Type`, strictly speaking objects that have a `type` method, and for which `Puppet::Resource::TypeCollection` has an `add_TYPE` method.

I can see how `instantiate_FunctionDefinition` does some dark magic to store function definitions in loaders - I am not sure where `instantiate_CapabilityMapping` would store its blueprints/annotations.

I've filed [APPMGMT-834](https://tickets.puppetlabs.com/browse/APPMGMT-834) to track the cross-file annotations issue to unblock this review; the functionality here is limited but still works as long as annotations are in the same file as the defined type.